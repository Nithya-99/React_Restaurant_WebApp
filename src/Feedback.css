.form-login{
	width:360px;
	padding: 10% 0 0;
	margin:auto;
}
.form{
	/*position: relative;*/
	top:100px;
	z-index: 9;
	max-width: 500px;
	margin: 0 auto 100px;
	padding: 45px;
	color: black;
}

.form input{
	font-family: 'Roboto', sans-serif;
	outline: 1;
	background:#f2f2f2;
	width: 100%;
	border:2;
	margin:0 0 15px;
	padding: 15px;
	box-sizing: border-box;
	font-size: 14px;
}
.form button{
	font-family: 'Roboto', sans-serif;
	text-transform: uppercase;
	outline: 0;
	background: #9e5343;
	width: 100%;
	border:1;
	padding: 15px;
	color: #ffffff;
	font-size: 14px;
	cursor: pointer;
    margin-top: 10px;
    margin-bottom: 20px;
}
.form button:hover, .form button:active{
	border-color: #9e5343;
	box-shadow: 1px 1px 10px 5px #b1604f;
    color: white;
}


.form .form1{
	background: #e7d46785;
    /* opacity: 0.8; */
	padding: 20px;

}
.error{
	color: red;
	font-weight: bold;
    font-size: small;
}
input[type=text], input[type=password],input[type=number], textarea, select{ 
	width: 100%; 
	padding: 12px 20px; 
	margin: 8px 0; 
	display: inline-block; 
	box-sizing: border-box; 
}