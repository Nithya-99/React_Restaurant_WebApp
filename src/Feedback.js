import { Component, useState } from "react"
import './Feedback.css'
import 'bootstrap/dist/css/bootstrap.min.css';
import swal from "sweetalert";

export default function Feedback() {
    const [email, setEmail] = useState('');
    const [message, setMessage] = useState('');

    const [eEmail, setEEmail] = useState('');
    const [eMessage, setEMessage] = useState('');

    const [eColor, setEColor] = useState('');
    const [mColor, setMColor] = useState('');

    let [eFlag, setEFlag] = useState(false);
    let [mFlag, setMFlag] = useState(false);

    function validate(event) {
        if (message == '') {
            setMColor('red');
            setEMessage('Please enter something!');
            setMFlag(false);
        } else {
            setMColor('#55d83b');
            setEMessage('');
            setMFlag(true);
        }

        if (/^[a-zA-Z0-9]+@[a-zA-Z0-9]+\.[A-Za-z]+$/.test(email) && email != '') {
            setEEmail('');
            setEColor('#55d83b');
            setEFlag(true)
        } else {
            setEEmail('Invalid Email');
            setEColor('red');
            setEFlag(false)
        }
        if (mFlag && eFlag) {
            
            submit();
        }

        function submit() {
            setEFlag(false);
            setMFlag(false);
            setEColor('');
            setMColor('');
            setMessage('');
            setEmail('');
            document.getElementById('email').value = '';
            document.getElementById('message').value = '';
            swal({
                title: "Feedback submitted successfully",
                text: "Thankyou!",
                icon: "success",
                button: "OK",
            });
        }
    }

    return (
        <div className='form'>
            <form className='form1'>
                <h3 align='center'>Feedback</h3>
                <div className="container">

                    <input type="text" placeholder="Enter email" name="email" id="email" data-testid="email-input" onChange={event => setEmail(event.target.value)} style={{ borderColor: eColor }} />
                    <p className='error'>{eEmail}</p>

                    <textarea placeholder="Message" name="message" id="message" data-testid="message-input" onChange={event => setMessage(event.target.value)} style={{ borderColor: mColor }}>
                    </textarea>
                    <p className='error'>{eMessage}</p>

                    <div className="clearfix">
                        <button className="btn btn-lg" name="feedback" type="button"  data-testid="submit-feedback" onClick={() => validate()}>Submit</button>
                    </div>
                </div>
            </form>
        </div>
    )

}

