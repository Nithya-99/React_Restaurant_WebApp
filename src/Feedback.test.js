import React from "react";
import { render, fireEvent, cleanup, wait } from "@testing-library/react";
import Feedback from "./Feedback";
import '@testing-library/jest-dom/extend-expect';

describe('Test Feedback', ()=>{
    it('renders properly', ()=>{
        const {baseElement} = render(<Feedback/>);
        expect(baseElement).toBeInTheDocument();
    });

    it('receives input text', ()=>{
        const {getByTestId} = render(<Feedback/>);

        const email = getByTestId('email-input');
        expect(email).toBeInTheDocument();

        const message = getByTestId('message-input');
        expect(message).toBeInTheDocument();

        const feedback = getByTestId('submit-feedback');
        expect(feedback).toBeInTheDocument();
    })

    it('matches snapshot', ()=> {
        const {baseElement} = render(<Feedback/>);
        expect(baseElement).toMatchSnapshot();
    });
});
  