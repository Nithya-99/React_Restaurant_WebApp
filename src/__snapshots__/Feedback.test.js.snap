// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Feedback matches snapshot 1`] = `
<body>
  <div>
    <div
      class="form"
    >
      <form
        class="form1"
      >
        <h3
          align="center"
        >
          Feedback
        </h3>
        <div
          class="container"
        >
          <input
            data-testid="email-input"
            id="email"
            name="email"
            placeholder="Enter email"
            type="text"
          />
          <p
            class="error"
          />
          <textarea
            data-testid="message-input"
            id="message"
            name="message"
            placeholder="Message"
          />
          <p
            class="error"
          />
          <div
            class="clearfix"
          >
            <button
              class="btn btn-lg"
              data-testid="submit-feedback"
              name="feedback"
              type="button"
            >
              Submit
            </button>
          </div>
        </div>
      </form>
    </div>
  </div>
</body>
`;
